Tools/autotest/arduplane.py

```diff
@@ -1765,6 +1765,109 @@ def fly_terrain_mission(self):
 
         self.fly_mission("ap-terrain.txt", mission_timeout=600)
 
+    def ekf_lane_switch(self):
+
+        # new lane swtich available only with EK3
+        self.set_parameter("EK3_ENABLE", 1)
+        self.set_parameter("EK2_ENABLE", 0)
+        self.set_parameter("AHRS_EKF_TYPE", 3)
+        self.set_parameter("EK3_AFFINITY", 15) # enable affinity for all sensors
+        self.set_parameter("EK3_IMU_MASK", 3) # use only 2 IMUs
+        self.set_parameter("GPS_TYPE2", 1)
+        self.set_parameter("SIM_GPS2_DISABLE", 0)
+        self.set_parameter("SIM_BARO2_DISABL", 0)
+        self.set_parameter("SIM_BARO_COUNT", 2)
+        self.set_parameter("ARSPD2_TYPE", 2)
+        self.set_parameter("ARSPD2_USE", 1)
+        self.set_parameter("ARSPD2_PIN", 2)
+        self.set_parameter("SIM_SPEEDUP", 20)
+        self.set_parameter("SIM_RATE_HZ", 200)
+
+        # some parameters need reboot to take effect
+        self.reboot_sitl()
+        
+        # get flying
+        self.takeoff(alt=25)
+        self.change_mode('CIRCLE')
+
+        self.lane_switches = []
+
+        # add a EKF lane switch hook
+        def statustext_hook(mav, message):
+            if message.get_type() != 'STATUSTEXT':
+                return
+            # example msg: EKF3 lane switch 1
+            if not message.text.startswith("EKF3 lane switch "):
+                return
+            newlane = int(message.text[-1])
+            self.lane_switches.append(newlane)
+
+        self.install_message_hook(statustext_hook)
+        
+        self.context_push()
```

Comment:
`self.context_push()` should be at the start of the test before starting to modify the parameter: it say : starting from there I save the parameter modification. That will allow to have `self.context_pop()` at the end of the test that rewind the parameter change and let the state of SITL clean for further test!